### PR TITLE
Update dockerfile

### LINKED ISSUE #110: Update Docker Compose configuration file
The [Docker Compose file](https://github.com/creativecommons/vue-vocabulary/blob/master/Dockerfile) for the project, written in the initial days of the project has now gone out of sync with the capabilities and features of the project. The goal is to update the file in accordance with the current state and eventually push for it to the the standard development environment for the project.

### PR DESCRIPTION
**Pull request by jshreyans**
**Fixes**

Fixes #110  by @dhruvkb 

**Description**
Updates the Dockerfile to support a stable Docker build. Currently, it serves the Vocabulary website.

**Type of PR** 
This PR is a refactor.

**Technicalities**

**Tests**

**Screenshots**

**Checklist:**
 
- [x] My pull request has a descriptive title (not a vague title like "Update `index.md`").
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

**Legal**
By submitting this PR, I agree to abide by the terms of the Developer 
Certificate of Origin.

<details>
<summary>Developer Certificate of Origin</summary>

Developer Certificate of Origin<br>
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.<br>
1 Letterman Drive<br>
Suite D4700<br>
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
</details>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+FROM node:10
 
 # Make a codebase directory and work in it
 RUN mkdir codebase
 WORKDIR /codebase
 
 # Load the package.json and package-lock.json files
-COPY package.json .
-COPY package-lock.json .
+COPY package*.json ./
```
Comment (dhruvkb) [inline Dockerfile]: [Explicit is better than implicit. Simple is better than complex.](https://www.python.org/dev/peps/pep-0020/#the-zen-of-python)

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 
 # Install packages
 RUN npm install
+RUN npm install -g @vue/cli
```
Comment (dhruvkb) [inline Dockerfile]: No need for a separate install of the Vue CLI, it is automatically installed by the previous command.

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+
+# Copy over all project files to container
+COPY . ./
+
+# Review needed
+EXPOSE 8080
+RUN npm run serve:website
```
Comment (dhruvkb) [inline Dockerfile]: `RUN` command is handled by Docker Compose

File: Dockerfile [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
 
 # Install packages
 RUN npm install
+RUN npm install -g @vue/cli
+
+# Copy over all project files to container
+COPY . ./
+
+# Review needed
+EXPOSE 8080
```
Comment (dhruvkb) [inline Dockerfile]: Expose 8080 (website), 6006 (storybook) and 6060 (styleguide).

File: Dockerfile [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
-COPY package.json .
-COPY package-lock.json .
+COPY package*.json ./
 
 # Install packages
 RUN npm install
+RUN npm install -g @vue/cli
+
+# Copy over all project files to container
+COPY . ./
```
Comment (dhruvkb) [inline Dockerfile]: This is not required as we're using Docker Compose to map the project files directly into a volume on the container.

File: Dockerfile [hunk 6]
```diff
@@ -1,13 +1,20 @@
 # Build on top of a Node.js + npm image.
-FROM node:latest
+FROM node:10
```
Comment (dhruvkb) [inline Dockerfile]:
```suggestion
FROM node:lts
```

**Comment by jshreyans** [comment]
@dhruvkb I've commited the updated Dockerfile with all the changes. Hope it looks fine now. 
Learnt a lot solving this issue. Thanks!

File: Dockerfile (hunk 1 above)
Comment (jshreyans) [reply Dockerfile]: > [Explicit is better than implicit. Simple is better than complex.](https://www.python.org/dev/peps/pep-0020/#the-zen-of-python)

I'll definitely remember this :smile:

**Comment by jshreyans** [comment]
@dhruvkb Are there any more changes needed here?

File: Dockerfile [hunk 7]
```diff
@@ -1,13 +1,19 @@
...
 
 # Make a codebase directory and work in it
 RUN mkdir codebase
 WORKDIR /codebase
 
 # Load the package.json and package-lock.json files
-COPY package.json .
-COPY package-lock.json .
+COPY package.json ./
+COPY package-lock.json ./
```
Comment (dhruvkb) [inline Dockerfile]: What effect does adding the / have on the image?

**Comment by dhruvkb** [comment]
@jshreyans the work on the `Dockerfile` is commendable. But it still doesn't fix the issue #110 which was to 'Update **Docker Compose** configuration file'. Can you work on that in this PR?

File: Dockerfile (hunk 7 above)
Comment (jshreyans) [reply Dockerfile]: On some Linux distros, using just the dot throws an error. I remember this from when I was trying out [Manjaro](https://manjaro.org/) a few months back. Hence, I added the slashes as a contingency here

**Comment by jshreyans** [comment]
I'm really sorry for the delay here. Will surely post changes in the docker-compose file asap.

**Comment by jshreyans** [comment]
@dhruvkb I've made changes to the docker-compose file.  
There are 3 services defined in the file:
  - website
  - styleguide 
  - storybook

Any of these services can be deployed in its own container with the command
```
docker-compose up <service_name>
```
Apart from this, I've made some other small changes. Looking forward to your reviews!

File: Dockerfile (hunk 7 above)
Comment (dhruvkb) [reply Dockerfile]: Oh, okay. That's nice.

**Pull request merged**